react-main/src/app/components/Login/layout.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { reduxForm, Field } from 'redux-form';
+import styles from './styles.scss';
+
+export const minLength = min => value =>
+  value && value.length < min ? ` Must be ${min} characters or more` : undefined
+
+const minLength8 = minLength(8)
+
+const required = value => (value ? undefined : ' Required');
+
+const renderField = ({
+  input,
+  placeholder,
+  type,
+  meta: {touched, error}
+}) => (
+  <div>
+    <div>
+      <input {...input} placeholder={placeholder} type={type}/>
+      {touched && (error && <span>{error}</span>)}
+    </div>
+  </div>
+);
+
+const FormInput = props => {
+  const {handleSubmit} = props;
+  return (
+    <form onSubmit={handleSubmit}>
+      <div>
+        <label>Email</label>
+        <div>
+          <Field
+            name="email"
+            component={renderField}
+            type="email"
+            placeholder="example@gmail.com"
+            validate={required}
+           />
+        </div>
+        <label>Password</label>
+        <div>
+          <Field
+            name="password"
+            component={renderField}
+            type="password"
+            validate={[required, minLength8]}
```

Comment:
are you sure this validation works? Usually, the validate prop needs to have a fixed reference to work, and each time this is render it's creating a new array